### PR TITLE
chore: archiecture page updates

### PR DESCRIPTION
**Pull request by jonathannorris**

### CODE REVIEW

**Comment by jonathannorris** [comment]
* **#652** <a href="https://app.graphite.dev/github/pr/DevCycleHQ/devcycle-docs/652?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @jonathannorris and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: docs/introduction/architecture.md [hunk 1]
```diff
@@ -19,103 +19,106 @@ need to implement Server-Sent Event (SSE) connections fully.
...
 
-1. Features created in the Dashboard interact directly with our public [Management API](/management-api/), 
+1. Features created in the Dashboard interact directly with our public **[Management API](/management-api/)**, 
 to create and manage various models like Audiences / Features / Variables / Variations / etc.
 
-2. Changes made by the Management API are delivered via a Queue to the Config Service. 
+2. Changes made by the **Management API** are delivered via a queue to the **Config Service**. 
 
-3. The Config Service bundles all changes into a JSON configuration for each project, environment, 
+3. The **Config Service** bundles all changes into a JSON configuration for each project, environment,
```
Comment (ajwootto) [inline docs/introduction/architecture.md:27]: I think technically we capitalize Project and Environment

File: docs/introduction/architecture.md [hunk 2]
```diff
@@ -19,103 +19,106 @@ need to implement Server-Sent Event (SSE) connections fully.
...
-However, WASM is not a silver bullet, and for certain very highly threaded low-level use cases, 
+However, WASM is not a silver bullet, and for certain very highly threaded lower-level use cases, 
 we have built a native implementation, for example, in our [GO SDK](https://github.com/DevCycleHQ/go-server-sdk).
 
 ## Local Bucketing Server SDK Architecture
 
 ![Architecture Docs Diagrams - Local Server SDK.svg](/architecture-docs-diagrams-local-server-sdk.svg)
 
-**Note: Server SDKs currently do not support Server-Sent Events for updates to the config.**
+**Note: Server-sent event support for Server SDKs is currently under development.**
```
Comment (JamieSinn) [inline docs/introduction/architecture.md:68]:
```suggestion
**Note: Server-sent event support for Server SDKs is currently under active development.**
```

File: docs/introduction/architecture.md [hunk 3]
```diff
@@ -19,103 +19,106 @@ need to implement Server-Sent Event (SSE) connections fully.
...
 ## Cloud Bucketing Server SDK Architecture
 
 ![Architecture Docs Diagrams - Cloud Server SDK.svg](/architecture-docs-diagrams-cloud-server-sdk.svg)
 
-**Note: Server SDKs currently do not support Server-Sent Events for updates to the config.**
+For most use cases, local bucketing SDKs provide superior performance and reliability.
+However, the cloud-bucketing SDKs can make integration easier for specific use cases where access to
+[EdgeDB](/extras/edgedb) to integrate user data between client-side and backend applications is needed.
+
+**Note: Server-sent event support for Server SDKs is currently under development.**
```
Comment (ajwootto) [inline docs/introduction/architecture.md:87]: whys this note repeated in this section? server sent events arent relevant for cloud bucketing anyway

File: docs/introduction/architecture.md [hunk 4]
```diff
@@ -19,103 +19,106 @@ need to implement Server-Sent Event (SSE) connections fully.
...
 we have a cross-platform end-to-end SDK test-harness to ensure platform consistency.
 
-- **Secure**: WASM runs in a [memory-safe sandboxed execution environment](https://webassembly.org/docs/security/) 
-that has been proven to be secure over many years.
+- **Secure**: WASM runs in a [memory-safe sandboxed execution environment](https://webassembly.org/docs/security/)
+that has proven its security credentials across web browsers.
 
-However, WASM is not a silver bullet, and for certain very highly threaded low-level use cases, 
+However, WASM is not a silver bullet, and for certain very highly threaded lower-level use cases, 
 we have built a native implementation, for example, in our [GO SDK](https://github.com/DevCycleHQ/go-server-sdk).
```
Comment (JamieSinn) [inline docs/introduction/architecture.md:62]:
```suggestion
However, WASM is not a silver bullet, and for certain very higher performance/concurrent threading use cases, 
we have built a native implementation in our [GO SDK](https://github.com/DevCycleHQ/go-server-sdk).
```

File: docs/introduction/architecture.md [hunk 5]
```diff
@@ -19,103 +19,106 @@ need to implement Server-Sent Event (SSE) connections fully.
...
+2. The Client SDK API calls the shared bucketing and segmentation library to combine user, device, 
+and configuration data to bucket the user into features and variations to determine variable values. 
+This data is returned to the SDKs to be cached locally and used each time `variableValue()` / `variable()` is called.
 
-3. When user data is updated, using the `identifyUser()` or `resetUser()` methods, the SDKs will 
+3. When user data is updated using the `identifyUser()` or `resetUser()` methods, the SDKs will 
 request a new configuration from the Client SDK API.
 
-4. The Client SDKs make SSE connections to receive real-time configuration data updates.
+4. The Client SDKs make SSE connections to receive real-time updates, which triggers a request for an
```
Comment (ajwootto) [inline docs/introduction/architecture.md:111]:
```suggestion
4. The Client SDKs make SSE connections to receive real-time updates, which trigger a request for an 
```

File: docs/introduction/architecture.md [hunk 6]
```diff
@@ -19,103 +19,106 @@ need to implement Server-Sent Event (SSE) connections fully.
...
 
-5. Event data is sent on an interval to our Events API.
+5. Event data is sent at intervals to our Events API.
 
 ## SDK Test Harness
 
 Our [SDK Test Harness](https://github.com/DevCycleHQ/test-harness) aims to define a set of standardized end-to-end tests 
-that run against multiple SDKs written in different languages. The tests defined using Jest
-are run using a set of HTTP requests made to a series of proxy servers for each language of the SDKs.
+that run against multiple SDKs written in different languages. The tests defined using Jest are run using
```
Comment (ajwootto) [inline docs/introduction/architecture.md:119]:
```suggestion
that run against multiple SDKs written in different languages. The tests are defined using Jest, and are run using 
```

File: docs/introduction/architecture.md [hunk 7]
```diff
@@ -19,103 +19,106 @@ need to implement Server-Sent Event (SSE) connections fully.
...
+5. Changes to the configuration data trigger **Cloudflare's CDN** cache invalidation, 
 which can serve new data in approximately ~1 second globally.
 
 6. Finally, an update notification is pushed to all connected SDKs via a server-sent event (SSE) connection
 that informs them that new configuration data updates are available.
 
 ## Shared Bucketing and Segmentation Library
 
-The shared bucketing and segmentation library is the core of our SDKs and APIs logic. It combines configuration data 
+The shared bucketing and segmentation library is the core of our SDKs and API logic. It combines configuration data
```
[the remark this answers is not in the document]
Comment (jonathannorris) [reply docs/introduction/architecture.md:40]: its used across multiple APIs.

[the remark this answers is not in the document]
Comment (JamieSinn) [reply docs/introduction/architecture.md:40]: Maybe clarify that? having the management api right before this is a bit confusing?

**Review by suthar26 — APPROVED** [review]

**Pull request merged**